lib/ReorderBasicBlocksPass.cpp

```diff
@@ -27,11 +33,18 @@ struct ReorderBasicBlocksPass : public FunctionPass {
 
   void getAnalysisUsage(AnalysisUsage &AU) const override {
     AU.addRequired<DominatorTreeWrapperPass>();
+    AU.addRequired<LoopInfoWrapperPass>();
   }
 
   bool runOnFunction(Function &F) override;
+
+private:
+  // Produce structured sorting from |block| into |order|.
+  void StructuredOrder(BasicBlock *block, const LoopInfo &LI,
```

Comment:
Should say what structured order is, i.e. what properties it has (not how you compute it).  And maybe why it's nice.

cfg.h from SPIRV-Tools says:


    // This order has the property that dominators come before all blocks they
    // dominate and merge blocks come after all blocks that are in the control
    // constructs of their header.

It should also add the properties of the continue construct as well.... :-)
I think that would be
    // ... and the continue construct blocks appear after the blocks in the associated loop construct.

So:
    // This order has the following properties:
    // -  dominators come before all blocks they dominate
    // - a merge blocks follows all blocks that are in the control
    //   constructs of the associated header.
    // - blocks in a loop construct precede the blocks in the associated continue construct
    // - no interleaving: blocks in one construct A are mixed with blocks in another construct B only if one is nested inside the other (without loss of generality, A's header dominates all blocks in B, and all B's blocks appear contiguously within A's blocks.)